### ISSUE TITLE
[Feature Request] Support line styles in Scattermapbox

### ISSUE THREAD

**Issue by Worschi** [plotly/plotly.js#6791]
It would be nice to plot lines of different styles, e.g. dashed, dotted, etc. to a Scattermapbox layer. The lack for this feature has been noted here [How do I change the line style to dash in a scattermapbox plot?](https://community.plotly.com/t/how-do-i-change-the-line-style-to-dash-in-a-scattermapbox-plot/33469), and here [How to change line shape in mapbox maps?](https://community.plotly.com/t/how-to-change-line-shape-in-mapbox-maps/69687)

Currently (Plotly 5.18), the only [supported line attributes](https://plotly.com/javascript/reference/scattermapbox/#scattermapbox-line) are "color" and "width" . However, distinguishing lines by color is not sufficient for monochrome styled figures. In addition, certain line styles are frequently expected:
* minor roads = dashed line 
* under construction = dotted line

**Comment by Coding-with-Adam** [comment]
Good idea, @Worschi . w
Would you be interested in seeing if you can find this functionality in Mapbox itself and figure out how the API compares with what we have for line dash and scatter traces?

**Comment by Worschi** [comment]
Hello @Coding-with-Adam, thanks for noticing! I'd probably be able to do some more research and post my findings here. I already came accross a working code snippet for "line-dasharray" styled lines in the mapbox-gl.js class. 

Have a look at this [minimal jsfiddle example](https://jsfiddle.net/xgsuw671/). It uses a "line-dasharray" in the mapbox-gl.js to plot dashed lines. There may be better ways though since my initial reseach was not thorough.

Would you be so nice as to point me in the right direction for plotly API to mabox API mapping?

**Comment by archmoj** [comment]
Please see this mock: https://raw.githubusercontent.com/plotly/plotly.js/master/test/image/mocks/mapbox_geojson-attributes.json which renders as https://raw.githubusercontent.com/plotly/plotly.js/master/test/image/baselines/mapbox_geojson-attributes.png

**Comment by Worschi** [comment]
Nice! Thanks, @archmoj. I'll try this out beginning of the week and let you know if it worked for my use case. I'll keep you posted and close this issue in case it did.

**Comment by alexcjohnson** [comment]
Thanks @Worschi! Good to know that this can be done with a geojson layer. But I'd like to keep this issue open until we investigate adding it to scattermapbox, that will still be more useful if we can get it working.

**Comment by Worschi** [comment]
Hi @alexcjohnson, fine with me. I am not sure when/if I'll be able to investigate further. The team already moved on and, for now, settled with not having dashed lines.

**Comment by kussshhhh** [comment]
@alexcjohnson is this issue still open?

**Comment by Coding-with-Adam** [comment]
hi @kussshhhh 
Yes, this issue is still open

**Comment by kussshhhh** [comment]
hi @Coding-with-Adam 
i want to help with this issue, where in the codebase do i have to add code to facilitate dotted and dash lines

**Comment by Ayfri** [comment]
Any update ?

**Comment by salignatmoandal** [comment]
is this issue still open?

**Comment by gvwilson** [comment]
yes, but I don't know when someone will be able to work on it - we have a very large backlog that we're trying to clear
